src/CMakeLists.txt

```diff
@@ -1043,76 +1043,40 @@ target_compile_options(H5detect
 )
 set (lib_prog_deps ${lib_prog_deps} H5detect)
 
-# check if a pregenerated H5Tinit.c file is present
-if (NOT EXISTS "${HDF5_GENERATED_SOURCE_DIR}/H5Tinit.c")
```

Comment:
This check is if we are cross-compiling with a pre-generated file - because the host system cannot generate a file for the target system. How is it handled in your changes?

Reply:
Hrm. This causes problems because rerunning the configure after the build happens causes it to detect the "pre-generated" file and confuses reconfigures. I think another approach for cross-compilations where the *source* directory gets the to-be-used file is used instead of overloading the existence of the output like this.

Reply:
If there is a cross-compiling emulator available (e.g., Wine when doing MinGW on Linux), it will be able to run the binary. So it's not just "cross compiling, therefore impossible".